src/client/pfs/pfs.proto

```diff
@@ -333,20 +326,6 @@ message PutFileRequest {
   // overwrite_index is the object index where the write starts from.  All
   // existing objects starting from the index are deleted.
   OverwriteIndex overwrite_index = 10;
-
-  // header indicates data indicates data that should be prepended to each
-  // written file (only used if delimiter is not NONE). This allows for sensible
-  // semantics when splitting e.g. .csv or .pgdump files, where reading any
-  // shard (as in a pipeline) will return the header data required to interpret
-  // that shard, but reading all shards (as with a glob pattern) will not repeat
-  // the header but only include it once at the beginning.
-  Header header = 11;
-
-  // footer is similar to 'header', but it indicates data indicates data that
-  // should be appended written file (again, only used if delimiter is not
-  // NONE). This allows for the same sensible semantics as 'header' but with
-  // respect to data that should appear at the end of any shard(s).
-  Footer footer = 12;
```

Comment:
So this means that header and footer are leaving the API explicitly and only accessible through `--split`? Is that permanent or temporary?

Reply:
I mean, in my view, it's hopefully permanent, though it ultimately depends on what customers ask for. I think supporting the operation where users convert non-header/footer directories into header/footer directories (or worse, supporting mixtures of header/non-header files or multiple headers) will be annoying and ultimately prohibitive.

Basically, I see this as the narrowest interface that solves the problem of splitting files with headers, and supporting narrow interfaces is easier than supporting wide ones.

Reply:
So I agree in general on supporting as narrow an interface as possible. But it seems like this puts an onus on us to add parsers for every file format people care about. We'd probably have to add a csv parser almost immediately, and I bet a bunch of other formats from there.

I agree that it does expose some kind of weird calling patterns though. It's reasonable if succussive `PutFile`s always have the same `header` set, but as soon as it's mix and match it gets dicey.

Reply:
So one change that I'm considering adding to this PR is a flag `--header-records=N`. Then if you set `--split` with anything other than `SQL` (which has its own complicated header logic) it just grabs the first N records and makes them into a header. That would work for most record-based parsers (`line`, `csv`, `json`, etc). Thoughts?

Reply:
Also, re: CSV parsing in particular, ask and ye shall receive: https://golang.org/pkg/encoding/csv/

Reply:
Yeah, that seems like a great feature that covers a lot of bases, and shouldn't be too hard to implement.